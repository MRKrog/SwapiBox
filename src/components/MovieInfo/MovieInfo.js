import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './MovieInfo.scss';


class MovieInfo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      movieCrawl: '',
      movieTitle: '',
      movieRelease: ''
    }
  }

  componentDidMount() {
    this.getData();
  }

  getData = () => {
    return fetch(`https://swapi.co/api/films/${this.props.movieNumber}`)
    .then(response => response.json())
    .then(fetchData => {
      this.setState({
        movieCrawl: fetchData.opening_crawl,
        movieTitle: fetchData.title,
        movieRelease: fetchData.release_date
      });
    })
    .catch(error => {
      throw new Error(error);
    });
  }

  render() {
    const { movieCrawl, movieTitle, movieRelease } = this.state;
    return (
        <div className="MovieInfo">
          <div className="fade"></div>
          <section className="star-wars">
            <div className="crawl">
              <div className="title">
                <p>{movieTitle}</p>
                <h1>{movieRelease}</h1>
              </div>
              <p>{movieCrawl}</p>
            </div>
          </section>
          <button className="startBtn" onClick={this.props.handleUser}>Explore Movie</button>
        </div>
      )
    }
}

MovieInfo.propTypes = {
  movieNumber: PropTypes.number.isRequired
}

export default MovieInfo;
