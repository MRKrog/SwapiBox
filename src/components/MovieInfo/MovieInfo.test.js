import React from 'react';
import MovieInfo from './MovieInfo';
import { shallow } from 'enzyme';

// import { fetchAnything } from '../Fetch/fetchAnything.js';
// jest.mock('../Fetch/fetchAnything.js');

const handleUser = jest.fn();

describe('MovieInfo', () => {
  let movieInfoComponenet;
  let mockData;
  let movieNumber = 6;

  beforeEach(() => {

    movieInfoComponenet = shallow(
      <MovieInfo handleUser={handleUser}
                 movieNumber={movieNumber}
      />)

    mockData = {
      title: "Revenge of the Sith",
      episode_id: 3,
      opening_crawl: "War! The Republic is crumbling nunder attacks by the ruthless Sith Lord",
      director: "George Lucas",
      producer: "Rick McCallum",
      release_date: "2005-05-19"
    }

    fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve(mockData)
    }))
  })


  it("should match snapshot when all data is passed correctly", () => {
    expect(movieInfoComponenet).toMatchSnapshot();
  });

  it("fetch call takes expected url", async () => {

    movieInfoComponenet.instance().getData();
    expect(movieInfoComponenet.state()).toEqual({
      movieCrawl: "War! The Republic is crumbling nunder attacks by the ruthless Sith Lord",
      movieTitle: "Revenge of the Sith",
      movieRelease: "2005-05-19"
    });
    expect(movieInfoComponenet).toMatchSnapshot();
  });


})
