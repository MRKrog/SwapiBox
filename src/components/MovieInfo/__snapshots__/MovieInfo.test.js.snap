// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo fetch call takes expected url 1`] = `
<div
  className="MovieInfo"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p>
          Revenge of the Sith
        </p>
        <h1>
          2005-05-19
        </h1>
      </div>
      <p>
        War! The Republic is crumbling nunder attacks by the ruthless Sith Lord
      </p>
    </div>
  </section>
  <button
    className="startBtn"
    onClick={[MockFunction]}
  >
    Explore Movie
  </button>
</div>
`;

exports[`MovieInfo should match snapshot when all data is passed correctly 1`] = `
<div
  className="MovieInfo"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p />
        <h1 />
      </div>
      <p />
    </div>
  </section>
  <button
    className="startBtn"
    onClick={[MockFunction]}
  >
    Explore Movie
  </button>
</div>
`;
